### ISSUE TITLE
Feature request: ndarray.norm(), ndarray.H

### ISSUE THREAD

**Issue by Jakob-Unfried** [google/jax#1852]
I have recently come to Jax from pyTorch because i need the holomorphic autodiff.
I want to request two features because they would improve the readability of my code:

1. calling `norm()` on tensors:
I want to use something like
```
import jax.numpy as np
a = np.array([[1, 2], [3, 4]])
a_norm = a.norm()
```
which should just return `np.linalg.norm(a)`

2. a shorthand for the hermitian conjugate
Just as there is a shorthand `a.T` for the transpose, i would like to have `a.H == a.conj().T`

Disclaimer: As i am new to Jax, it is very possible that i am unaware of a good reason these do not exist.
Unfortunately this also means that I do not feel up to the task of proposing implementations myself.

**Comment by shoyer** [comment]
`jax.numpy` strives to reproduce NumPy's API. Given that goal, I'm not sure adding these methods/attributes makes sense for us, given that it would be a deviation from NumPy.

My own opinion is that we should reserve extending NumPy's API in `jax.numpy` for cases where it *isn't possible* do something in NumPy (e.g., we added the `bfloat16` dtype), not merely for cases where we would prefer different syntactic sugar.

**Comment by hawkinsp** [comment]
I agree with Stephan: on the whole we'd prefer not to deviate from NumPy without a good reason.

I know PyTorch has a convention that uses methods on Tensor (e.g., `.norm()`); I think on the whole JAX prefers to use free functions (`jax.numpy.linalg.norm()`). One advantage of free functions is that they are an open set: anyone can define one. Methods on `DeviceArray` are a bit harder for users to add freely; I'd characterize them as a closed set.

Syntactically there doesn't seem to be much difference to me between `x.norm()` vs `norm(x)`, and in fact the latter is one fewer character to type, although it's more characters if you qualify the name.

For the hermitian conjugate, in some of our libraries we define:
```
def T(x): return np.swapaxes(x, -1, -2)
def H(x): return np.conj(T(x))
```
and then you can write:
```
H(x)
```

I wouldn't want to export that name, but you can easily do that in your own code.

What do you think?

**Comment by Jakob-Unfried** [comment]
@hawkinsp 

I agree, that what i requested is nowhere near useful enough to justify a deviation from the original numpy.

Thank you for the suggestions of defining `H(x)`
While i had thought about that, it somehow didn't occur to me that i don't have to use `.T` but can define a `T(x)` myself.

Thanks for taking the time =)
